packages/api/amplify_api/lib/src/connectivity_plus_platform.dart

```diff
@@ -16,12 +16,13 @@ class ConnectivityPlusPlatform extends ConnectivityPlatform {
   @override
   Stream<ConnectivityStatus> get onConnectivityChanged => Connectivity()
           .onConnectivityChanged
-          .map((ConnectivityResult connectivityResult) {
-        switch (connectivityResult) {
-          case ConnectivityResult.none:
-            return ConnectivityStatus.disconnected;
-          default:
-            return ConnectivityStatus.connected;
-        }
+          .map((List<ConnectivityResult> connectivityResults) {
+        // Check if any of the results indicate a connection
+        final isConnected = connectivityResults
+            .any((result) => result != ConnectivityResult.none);
```

Comment:
Question: What should the behavior be if the result contains ConnectivityResult.none AND ConnectivityResult.wifi? Is that possible?

___

nit: I think you can simplify this:
```dart
final isConnected = connectivityResults.contains(ConnectivityResult.none);
```
___

This is out of scope for this PR, but it seems like ConnectivityResult.bluetooth would be treated like an active connection. Same with ConnectivityResult.other. I am not sure if we want to treat those like an active connection.

Reply:
based on their doc it is not possible to have ConnectivityResult.none AND ConnectivityResult.wifi.

> In case of no connectivity, the list contains a single element of [ConnectivityResult.none]. Note also that this is the only case where [ConnectivityResult.none] is present.

Reply:
do you mean 
```
final isNotConnected = connectivityResults.contains(ConnectivityResult.none);
```
I think this may be less efficent as it checks all the elements in the list

Reply:
Yes, that is what I meant.

`.contains()` will exit at the first element that matches.

[Here](https://api.dart.dev/stable/3.3.3/dart-core/Iterable/contains.html) is the implementation.

```dart
  bool contains(Object? element) {
    for (E e in this) {
      if (e == element) return true;
    }
    return false;
  }
```

Reply:
right, this is the case where the list contains ConnectivityResult.none (it happens when the device is not conntected and the list supposed to only have one element and it is ConnectivityResult.none). I meant the senarios when the device is connected and so the list may have multiple elements and none of them is ConnectivityResult.none

Reply:
I see. Makes sense.